hadoop-hdfs-project/hadoop-hdfs-rbf/src/main/java/org/apache/hadoop/hdfs/server/federation/router/RouterRpcServer.java

```diff
@@ -824,6 +865,50 @@ <T> T invokeOnNs(RemoteMethod method, Class<T> clazz, IOException ioe,
     throw ioe;
   }
 
+  <T> T invokeOnNsAsync(RemoteMethod method, Class<T> clazz, IOException ioe,
+      Set<FederationNamespaceInfo> nss) throws IOException {
+    if (nss.isEmpty()) {
+      throw ioe;
+    }
+
+    asyncComplete(null);
+    Iterator<FederationNamespaceInfo> nsIterator = nss.iterator();
+    asyncForEach(nsIterator, (foreach, fnInfo) -> {
+      String nsId = fnInfo.getNameserviceId();
+      LOG.debug("Invoking {} on namespace {}", method, nsId);
+      asyncTry(() -> {
+        rpcClient.invokeSingle(nsId, method, clazz);
```

Comment:
Same as above.
```java
getRPCClient().invokeSingle(nsId, method, clazz);
```

Reply:
@KeeProMise Sir, Thanks for your reviewing and pointing out the problem,  i will fix it soonly.